CleanWeather/Scenes/FavouriteCities/FavourtieCitiesInteractor.swift

```diff
@@ -0,0 +1,73 @@
+//
+//  FavourtieCitiesInteractor.swift
+//  CleanWeather
+//
+//  Created by Marek Skrzelowski on 05/12/2019.
+//  Copyright © 2019 vandermesis. All rights reserved.
+//
+
+import Foundation
+
+protocol FavouriteCitiesInteractor {
+    func getCities()
+    func didSelectCity(id: String)
+}
+
+final class FavouriteCitiesInteractorImpl {
+
+    private var allCities = [City]()
+
+    private let presenter: FavouriteCitiesPresenter
+    private let worker: FavouriteCitiesWorker
+    private let router: FavouriteCitiesRouter
+
+    init(presenter: FavouriteCitiesPresenter,
+         worker: FavouriteCitiesWorker,
+         router: FavouriteCitiesRouter) {
+        self.presenter = presenter
+        self.worker = worker
+        self.router = router
+    }
+}
+
+extension FavouriteCitiesInteractorImpl: FavouriteCitiesInteractor {
+
+    func getCities() {
```

Comment:
you'll need to chain these 2 requests:

show spinner
worker.fetchAllCitiees
if error then show error and hide spinner
if success then worker.fetchFavourite
if error then show error and hide spinner
if success then you've got everything to present